src/widgets/dropdown.cpp

```diff
@@ -361,53 +361,58 @@ void ShowDropDownListAt(Window *w, const DropDownList *list, int selected, int b
 		if (auto_width) max_item_width = max(max_item_width, item->Width() + 5);
 	}
 
-	/* Check if the status bar is visible, as we don't want to draw over it */
-	int screen_bottom = GetMainViewBottom();
+	/* Scrollbar needed? */
 	bool scroll = false;
 
-	/* Check if the dropdown will fully fit below the widget */
-	if (top + height + 4 >= screen_bottom) {
-		/* If not, check if it will fit above the widget */
-		int screen_top = GetMainViewTop();
-		if (w->top + wi_rect.top > screen_top + height) {
-			top = w->top + wi_rect.top - height - 4;
-		} else {
-			/* If it doesn't fit above the widget, we need to enable a scrollbar... */
-			int avg_height = height / (int)list->Length();
```

Comment:
Seems to me that `avg_height`, `height` & others should be uints, then this cast wouldn't be necessary. Is that feasible?

Reply:
I changed it. Is the comment for height right now?

Reply:
looks good now, thanks